### PR TITLE
Loginpage

### PR DESCRIPTION
**Pull request by Shailseen**
### Loginpage finish
**In this branch have** 

- Authentication of User login
- Context for User data
- Popup toast for Login/Error.
- Different popup message for different errors.

https://user-images.githubusercontent.com/90495133/160666252-20c6fdae-a8c9-4c44-9c47-2ebde7055cb7.mp4

### CODE REVIEW

File: src/components/carousel/Carousel.jsx [hunk 1]
```diff
@@ -4,13 +4,12 @@ import { Link } from "react-router-dom";
...
 const Carousel = () => {
   const [categories, setCategories] = useState([]);
-  console.log("hsjbs");
   const fetchCategories = async () => {
     try {
       const response = await axios.get(`/api/categories`);
       setCategories(response.data.categories);
     } catch (error) {
-      console.log(error);
+      alert(error);
```
Comment (vishalg8454) [inline src/components/carousel/Carousel.jsx:12]: You can use the same Toast component used in other places in your code.

File: src/context/toast-context.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const ToastContext = createContext();
+
+const useToast = () => useContext(ToastContext);
+
+const ToastProvider = ({ children }) => {
+  const [toastVal, setToastVal] = useState({
+    isDisplay: "hidden",
+    select: "",
+    msg: "",
+  });
```
Comment (vishalg8454) [inline src/context/toast-context.js:14]: I get the use of the variables but the names could be better. You can improve it.

**Review by vishalg8454 — APPROVED** [review]
All looks good, just some suggestions.

File: src/components/carousel/Carousel.jsx (hunk 1 above)
Comment (Shailseen) [reply src/components/carousel/Carousel.jsx:12]: sure, i will do it later.

File: src/context/toast-context.js (hunk 2 above)
Comment (Shailseen) [reply src/context/toast-context.js:14]: okay.

**Review by sagargse — APPROVED** [review]
All looks Good to me You can go ahead.

**Pull request merged**